raft/raft.go

```diff
@@ -632,7 +632,7 @@ func stepLeader(r *raft, m pb.Message) {
 			r.id, r.raftLog.lastTerm(), r.raftLog.lastIndex(), r.Vote, m.From, m.LogTerm, m.Index, r.Term)
 		r.send(pb.Message{To: m.From, Type: pb.MsgVoteResp, Reject: true})
 	case pb.MsgSnapStatus:
-		if pr.State != ProgressStateSnapshot {
+		if pr == nil || pr.State != ProgressStateSnapshot {
```

Comment:
Yea. I think that is up to you. I just want to find a more consistent way to check the nil value.